packages/styled-base/types/index.d.ts

```diff
@@ -24,127 +23,88 @@ export {
   ObjectInterpolation
 } from '@emotion/serialize'
 
-export { ComponentSelector, Interpolation }
-
-type JSXInEl = JSX.IntrinsicElements
-
-export type WithTheme<P, T> = P extends { theme: infer Theme }
-  ? P & { theme: Exclude<Theme, undefined> }
-  : P & { theme: T }
+export { ComponentSelector, Interpolation, PropsOf, Omit }
 
 export interface StyledOptions {
   label?: string
   shouldForwardProp?(propName: string): boolean
   target?: string
 }
 
-export interface StyledComponent<InnerProps, StyleProps, Theme extends object>
-  extends React.SFC<InnerProps & StyleProps & { theme?: Theme }>,
+/**
+ * @typeparam ComponentProps  Props which will be included when withComponent is called
+ * @typeparam SpecificComponentProps  Props which will *not* be included when withComponent is called
+ */
+export interface StyledComponent<ComponentProps, SpecificComponentProps = {}>
```

Comment:
Type change:

`StyledComponent<InnerProps, StyleProps, Theme extends object>` to `StyledComponent<ComponentProps, SpecificComponentProps = {}>`

Theme no longer explicitly a generic parameter on StyledComponent, simply use `StyledComponent<{ theme: Theme }>` instead. The second generic argument (`SpecificComponentProps`) are props which will not transfer to the next component when `withComponent` is called.

Migration example:

`StyledComponent<Props, {}, MyTheme>` -> `StyledComponent<Props & { theme?: MyTheme }>`